zapcore/level.go

```diff
@@ -55,6 +55,21 @@ const (
 	_maxLevel = FatalLevel
 )
 
+// UnmarshalLevel unmarshals a level based on the lower-case or all-caps ASCII
+// representation of the log level. If the provided ASCII representation is
+// invalid an error is returned.
+//
+// This is particularly useful when dealing with text input to configure log
+// levels.
+func UnmarshalLevel(text string) (Level, error) {
+	level := DebugLevel
+	err := level.UnmarshalText([]byte(text))
+	if err != nil {
+		return level, err
+	}
+	return level, nil
```

Comment:
minor change: we can simplify this:
```suggestion
	return level, err
```